src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+----
+====
+
+`VaultKeyValueOperations` supports all Key-Value operations such as `put`, `get`, `delete`, `list`.
+
+You can also interact with the specifics of the versioned key-value API. This is useful if you want to obtain a specific secret or you need access to the metadata.
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/KeyValueV2.java[tags=versionedApi]
+----
+<1> Store secrets at `elvis` in that is available under the `secret/` mount.
+<2> Storing data in the versioned backend returns metadata such as the version number.
+<3> The versioned Key-Value API allows retrieval of specific versions identified by the version number.
+<4> Versioned key-value secrets can be mapped into value objects.
+<5> When updating versioned secrets using CAS, the input must refer to the previously obtained version.
+====
+
+While using the `kv` v2 secrets engine through `VaultTemplate` is possible.
+It’s not the most convenient approach since the API offers a different approach to context paths and how input/output is represented.
+Specifically, interaction with the actual secrets requires wrapping/unwrapping of the data section and introducing a `data/` path segment between the mount and the secrets key.
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/KeyValueV2.java[tags=vaultOperations]
+----
+====
+
+You can find more details about the https://www.vaultproject.io/api-docs/secret/kv/kv-v2[Vault Key-Value version 2 API] in the Vault reference documentation.
+
+[[vault.core.backends.pki]]
+== PKI (Public Key Infrastructure)
+
+The `pki` secrets engine represents a backend for certificates by implementing certificate authority operations.
+
+The PKI secrets engine generates dynamic X.509 certificates.
+With this secrets engine, services can get certificates without going through the usual manual process of generating a private key and CSR, submitting to a CA, and waiting for a verification and signing process to complete.
+Vault's built-in authentication and authorization mechanisms provide the verification functionality.
+
+Spring Vault supports issuing, signing, revoking certificates, and CRL retrieval through `VaultPkiOperations`.
+All other PKI functionality can be used through `VaultOperations`.
+
+The following examples explain briefly the use of how to issue and revoke certificates:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/PKI.java[tags=pkiApi]
+----
+<1> Construct a certificate request using the `VaultCertificateRequest` builder.
+<2> Request a certificate from Vault.
+Vault acts as certificate authority and responds with a signed X.509 certificate.
+The actual response is a `CertificateBundle`.
+<3> Generated certificates can be loaded directly into a Java `KeyStore` containing public and private keys as well as the issuer certificate. `KeyStore` has a wide range of uses which makes this format suitable to configure for example a HTTP client, a database driver or a SSL-secured HTTP server.
```

Comment:
Change to "<3> You can load generated certificates directly into a Java `KeyStore` that contains public and private keys as well as the issuer certificate. `KeyStore` has a wide range of uses, which makes this format suitable to configure (for example a HTTP client, a database driver, or an SSL-secured HTTP server)."